### PR TITLE
refactor some functions and move duplicate to AbstractRpcRemotingClient classes.

### PR DESCRIPTION
**Pull request by isudox**
### Ⅱ. Does this pull request fix one issue?
This PR just do something like code optimization.

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
It has not change the code logic and feature, so there is no need to add more test cases.

### CODE REVIEW

**Comment by isudox** [comment]
The conflicts have been resolved. @slievrly

File: core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingClient.java [hunk 1]
```diff
@@ -259,7 +259,7 @@ public void shutdown() {
             }
             super.destroy();
         } catch (Exception exx) {
-            LOGGER.error("shutdown error:" + exx.getMessage());
+            LOGGER.error("shutdown error: {}", exx.getMessage());
```
Comment (jovany-wang) [inline core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingClient.java]: `Failed to shutdown: {}`

File: core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingClient.java [hunk 2]
```diff
@@ -321,6 +333,27 @@ public void dispatch(long msgId, ChannelHandlerContext ctx, Object msg) {
         }
     }
 
+    protected void reconnect(String transactionServiceGroup) {
+        List<String> availList = null;
+        try {
+            availList = getAvailServerList(transactionServiceGroup);
+        } catch (Exception exx) {
+            LOGGER.error(exx.getMessage());
```
Comment (jovany-wang) [inline core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingClient.java]: `Failed to get xxxx: {}`

Comment (isudox) [reply core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingClient.java]: Sorry I'm a little confused, but I changed my forked branch code and committed again.

**Comment by slievrly** [comment]
@isudox you can contact me, let us communicate quickly through dingtalk。

File: spring/src/main/java/com/alibaba/fescar/spring/annotation/GlobalTransactionScanner.java [hunk 3]
```diff
@@ -60,7 +60,7 @@
     private static final int ORDER_NUM = 1024;
     private static final int DEFAULT_MODE = AT_MODE;
 
-    private static final Set<String> PROXYED_SET = new HashSet<>();
+    private static final Set<String> PROXIED_SET = new HashSet<>();
```
Comment (slievrly) [inline spring/src/main/java/com/alibaba/fescar/spring/annotation/GlobalTransactionScanner.java]: why PROXYED-> PROXIED?

Comment (isudox) [reply spring/src/main/java/com/alibaba/fescar/spring/annotation/GlobalTransactionScanner.java]: @slievrly Already reverted.

File: core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingClient.java (hunk 2 above)
Comment (isudox) [reply core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingClient.java]: Resolved.

**Comment by a364176773** [comment]
由于前期Seata社区治理规范问题部分代码作者未签署CLA，可能引发社区知识产权风险问题。请所有在Seata社区贡献过代码（包含：主项目、官网、samples和多语言项目等）的 contributor 帮忙在这个链接登录github账号签署相应的开发者CLA，https://cla-assistant.io/seata/seata  。2023.1.31 未签署CLA的代码将会被rewrite，拜托大家帮忙签一下。

**Pull request merged**